File: Decorator/Java/DecoratorCoffee/src/VanillaDecorator.java
public class VanillaDecorator extends BaseDecorator{
    public VanillaDecorator(ICoffee coffee) {
        super(coffee);
    }

    @Override
    public double getCost() {
        return super.getCost() + 0.75;
    }

    @Override
    public String getDescription() {
        return super.getDescription() + ", vanilla";
    }
}

File: Decorator/DecoratorCoffee/DecoratorCoffee/VanillaDecorator.cs
namespace DecoratorCoffee
{
    public class VanillaDecorator : BaseDecorator
    {
        public VanillaDecorator(ICoffee coffee) : base(coffee)
        {
        }

        public override double GetCost() => base.GetCost() + 0.75;
        public override string GetDescription() => base.GetDescription() + ", vanilla";
    }
}
